QHyper/hyperparameter_gen/parser.py

```diff
@@ -123,18 +123,32 @@ def tmp(left: Any, right: Any, multiplier: int = 1) -> list[Any]:
 
 
 class Expression:
-    def __init__(self, polynomial: sympy.core.Expr) -> None:
-        self.polynomial: sympy.core.Expr = polynomial
+    def __init__(self, equation: sympy.core.Expr | dict) -> None:
+        if isinstance(equation, sympy.core.Expr):
+            self.polynomial: sympy.core.Expr | None = equation
+            self.dictionary: dict = self.as_dict()
+        elif isinstance(equation, dict):
+            self.polynomial: sympy.core.Expr | None = None
+            self.dictionary: dict = equation
+        else:
+            raise Exception(
+                f"Expression equation must be an instance of \
+                    sympy.core.Expr or dict, got of type: \
+                        {type(equation)} instead"
```

Comment:
Don't use \ to split lines . There are many better ways to do that 
```suggestion
                "Expression equation must be an instance of "
                "sympy.core.Expr or dict, got of type: "
                f"{type(equation)} instead"
```